src/picterra/tracer_client.py

```diff
@@ -288,3 +292,184 @@ def list_plots_analyses(
+            plots_analysis_id: id of the plots analysis for which we want to list the reports
+
+        Returns: see https://app.picterra.ch/public/apidocs/plots_analysis/v1/#tag/reports/operation/getReportsList
+        """  # noqa[E501]
+        return self._return_results_page(
+            f"plots_groups/{plots_group_id}/analysis/{plots_analysis_id}/reports/"
+        )
+
+    def list_plots_analyses_report_types(
+        self,
+        plots_group_id: str,
+        plots_analysis_id: str
+    ):
+        """
+        List all the plots analyses report types the user can use (see create_plots_analysis_report)
+
+        Args:
+            plots_group_id: id of the plots group
+            plots_analysis_id: id of the plots analysis
+
+        Returns:
+            see https://app.picterra.ch/public/apidocs/plots_analysis/v1/#tag/reports/operation/getReportTypesForAnalysis
+        """  # noqa[E501]
+        resp = self.sess.get(
+            self._full_url(f"plots_groups/{plots_group_id}/analysis/{plots_analysis_id}/reports/types/")
+        )
+        _check_resp_is_ok(resp, "Couldn't list report types")
+        return resp.json()
+
+    def create_plots_analysis_report_precheck(
+        self,
+        plots_group_id: str,
+        plots_analysis_id: str,
+        report_name: str,
+        plot_ids: List[str],
+        report_type: str,
+        metadata: Optional[dict] = None
+    ) -> dict:
+        """
+        Check creation of a report with the given parameters is ok
+
+        If the function fails, the report is not valid
+
+        Args:
+            plots_group_id: id of the plots group
+            plots_analysis_id: id of the plots analysis
+            report_name: name to give to the report
+            plot_ids: list of the plot ids to select for the report
+            report_type: type of report to generate, as per list_plots_analyses_report_types
+            metadata:  set of key-value pairs which may be included in the report
+
+        Returns:
+            dict: the precheck data
+        """
+        upload_id, upload_url = self._make_upload()
+        resp = requests.put(upload_url, data=json.dumps({"plot_ids": plot_ids}))
+        _check_resp_is_ok(resp, "Failure uploading plots file for analysis")
+        data = {
+            "name": report_name,
+            "upload_id": upload_id,
+            "report_type": report_type,
+            "metadata": metadata if metadata is not None else {}
+        }
+        resp = self.sess.post(
+            self._full_url(f"plots_groups/{plots_group_id}/analysis/{plots_analysis_id}/reports/precheck/"),
+            json=data
+        )
+        _check_resp_is_ok(resp, "Failure starting precheck")
+        self._wait_until_operation_completes(resp.json())
+        return {
+            "status": "passed",
+            "errors": {"critical": [], "high": []},
+            "critical_count": 0,
```

Comment:
Mmmh not sure we'll have the same errors as for analysis precheck. For now I'd just stick to status = passed so we're not limiting what we can do with errors here.